Fixes to window icons and click event

aurelion314 opened the pull request:
@Sellorio 

This gets it working for Dragonflight.

Changes include: 
- fixed click event (thanks to @Dannez83)
- updated xml template inheritance (@Dannez83 again, with tweaks by me)
- some lua tweaks and 2 new xml entries to go along with the new xml inheritance
- revert a caching feature that seemed to break the non global macro pages
- fix shift clicking on spell book buttons
- fix for popup scrollbar

@Dannez83, I've made this under a new fork because yours had altered line endings that made viewing the diff very difficult. I'm not sure how to correct that so I just started over.

Related Issue: #158

*Edit: Fixed the shift click issue and a scrollbar error. Should be fully usable now.

wing5wong commented on src/engine/mega-macro-engine.lua:
```diff
@@ -120,7 +120,7 @@ end
     return
         GenerateIdPrefix(macroId).."\n"..
-        "/click [btn:1] "..ClickyFrameName..macroId.."\n"..
+        "/click [btn:1] "..ClickyFrameName..macroId.." LeftButton\n"..
```
With this line like this, keybinds dont work

aurelion314 replied:
Can you let me know your client version? It works on mine, version 10.0.0.46313. 

A few people have mentioned this so I'm wondering if I'm missing a client update or if everyone else is >_>

wing5wong replied:
Same client as you have, playing in US realms

aurelion314 replied:
Thanks for checking. That's very strange though, because if I revert that line, it fails for me. I need the 'LeftButton' added.

I'm going to sleep on it for now.

Sellorio replied:
Once this is resolved I'll get this in and released :)

Dannez83 replied:
I just tried again with the new fork. It didn't work to start with. The commit didn't have the hook function commented out. Commenting out everything from line 739 onwards in mega-macro.window.lua and it works. 

The function seems to be for shift click adding spell names form the spell book. I'm guessing they've just updated the function name so will try to take a look at it later

jquick replied:
Works for me with no changes, live now, US.

Sellorio commented:
Looks like it's really close. I might hold off until a few of the uncertainties are resolved or if that looks to be taking a bit I'll merge in and create a prerelease version to help less-tech-savy players.

Dannez83 reviewed (commented):
SpellButton_OnModifiedClick has been replaced with a mixin. 
The below fixes the error with the missing function but needs further work to restore functionailty
`hooksecurefunc("SpellButton_OnModifiedClick", function(self)`
to
`hooksecurefunc(SpellButtonMixin, "OnModifiedClick", function(self)`

FrameXML/SpellBookFrame.lua line 708 for the new mixin location

morgan35543 commented:
For the modified click thing, it's on line 708: https://github.com/Gethe/wow-ui-source/blob/live/Interface/FrameXML/SpellBookFrame.lua

This is comparing it to a code snippet from here (4th comment) it's the same function: https://www.wowinterface.com/forums/showthread.php?t=36141

aurelion314 commented:
> SpellButton_OnModifiedClick has been replaced with a mixin.

This was a good find but I think the mixin acts as a template and by hooking the mixin, it's not hooking the thing that actually gets called. 

"SpellButton_OnModifiedClick" used to be listed in [here](https://github.com/Ketho/BlizzardInterfaceResources/blob/mainline_beta/Resources/FrameXML.lua), so I've checked the new file for something that replaces it but didn't find anything.

I'm a bit stumped at the moment, and unfortunately have to work on other things for a while. Hopefully someone finds something.

aurelion314 commented:
Hey @Sellorio, I think its good to go now.

I fixed the shift click issue. I also fixed the window.lua double load error that I noticed you mention in the comments ;) (It was already loading in the xml file, so it didn't need to be in the toc file)
I haven't heard anything else about the 'LeftButton' revert today, so I think it was a temporary issue on Blizzards side.

Hopefully my hacks to make it work aren't too hacky :)

Also, I didn't modify the readme like you mentioned, but if you want to throw @Dannez83 and my name in somewhere that's cool (I go by Cubelicious in game). Not a big a deal though.

Cheers

Sellorio commented:
> Hey @Sellorio, I think its good to go now.
> 
> I fixed the shift click issue. I also fixed the window.lua double load error that I noticed you mention in the comments ;) (It was already loading in the xml file, so it didn't need to be in the toc file) I haven't heard anything else about the 'LeftButton' revert today, so I think it was a temporary issue on Blizzards side.
> 
> Hopefully my hacks to make it work aren't too hacky :)
> 
> Also, I didn't modify the readme like you mentioned, but if you want to throw @Dannez83 and my name in somewhere that's cool (I go by Cubelicious in game). Not a big a deal though.
> 
> Cheers

Is that everything?

wing5wong commented:
Il test the reverted code tonight or can feedback to you after a release if you want to push ahead anyway.

Seems strange that 2 different group of people have different results - had this been confirmed with anyone else to be working that previously had the issue?

Regards,
Sean Anderson

On 28/10/2022, at 2:10 PM, Mike ***@***.***> wrote:

﻿

Hey @Sellorio<https://github.com/Sellorio>, I think its good to go now.

I fixed the shift click issue. I also fixed the window.lua double load error that I noticed you mention in the comments ;) (It was already loading in the xml file, so it didn't need to be in the toc file)
I haven't heard anything else about the 'LeftButton' revert today, so I think it was a temporary issue on Blizzards side.

Hopefully my hacks to make it work aren't too hacky :)

Also, I didn't modify the readme like you mentioned, but if you want to throw @Dannez83<https://github.com/Dannez83> and my name in somewhere that's cool (I go by Cubelicious in game). Not a big a deal though.

Cheers

—
Reply to this email directly, view it on GitHub<https://github.com/Sellorio/mega-macro/pull/162#issuecomment-1294287517>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ABBT5UOQOFNFGHZ3WRWDF6TWFMRWZANCNFSM6AAAAAARPWLDLU>.
You are receiving this because you commented.Message ID: ***@***.***>

aurelion314 commented:
Ok I figured it out. For some reason it depends on the value of the CVar "ActionButtonUseKeyDown". I'm pretty sure this is a bug on blizzards side, but I added a fix that checks the value of that and changes the click text based on it. Should work for everyone now.

@Sellorio  @wing5wong

wing5wong commented:
Confirming its working my end with this change

Thanks

From: ***@***.***>
Sent: Friday, 28 October 2022 4:58 pm
To: ***@***.***>
Cc: Sean ***@***.***>; ***@***.***>
Subject: Re: [Sellorio/mega-macro] Fixes to window icons and click event (PR #162)


Ok I figured it out. For some reason it depends on the value of the CVar "ActionButtonUseKeyDown". I'm pretty sure this is a bug on blizzards side, but I added a fix that checks the value of that and changes the click text based on it. Should work for everyone now.

@Sellorio<https://github.com/Sellorio> @wing5wong<https://github.com/wing5wong>

—
Reply to this email directly, view it on GitHub<https://github.com/Sellorio/mega-macro/pull/162#issuecomment-1294419741>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ABBT5UKI2O726K2IEWYWIHTWFNFMVANCNFSM6AAAAAARPWLDLU>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

Dannez83 commented:
> Ok I figured it out. For some reason it depends on the value of the CVar "ActionButtonUseKeyDown". I'm pretty sure this is a bug on blizzards side, but I added a fix that checks the value of that and changes the click text based on it. Should work for everyone now.
> 
> @Sellorio @wing5wong

Nice one @aurelion314, and on those other fixes too
